contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol

```diff
@@ -112,14 +116,18 @@ contract OCR2DROracle is OCR2DROracleInterface, OCR2Base {
     bytes[] memory errors;
     (requestIds, results, errors) = abi.decode(report, (bytes32[], bytes[], bytes[]));
     for (uint256 i = 0; i < requestIds.length; i++) {
-      fulfillRequest(requestIds[i], results[i], errors[i]);
-    }
-  }
-
-  modifier validateRequestId(bytes32 requestId) {
-    if (s_commitments[requestId].client == address(0)) {
-      revert InvalidRequestID();
+      try s_registry.concludeBilling(requestIds[i], results[i], errors[i], transmitter, initialGas) returns (
+        bool success,
+        uint96 /* cost */
+      ) {
+        if (success) {
```

Comment:
@pinebit this error handling got a little bit janky because try/catch doesn't decode custom errors.
We could do something like [this](https://github.com/ethereum/solidity/issues/11278#issuecomment-1004257852), but it might be too messy.